### ISSUE TITLE
Setting messageSendingQueueSupplier after construction results in messages not being sent

### ISSUE THREAD

**Issue by kumpelblase2** [KittehOrg/KittehIRCClientLib#246]
When setting the messageSendingQueueSupplier on a constructed client, messages that are not sent as immediate messages, won't get sent at all. This includes, for example, join messages. I can reproduce this using something like this:

```Kotlin
val builtClient = Client.builder().nick(twitchSettings.appUsername)
    .server().host(TWITCH_SERVER).port(TWITCH_PORT).password(serverPassword).secure(true).then()
            .build()
TwitchSupport.addSupport(builtClient, false)
````
and later:
```Kotlin
    @Handler
    fun onConnected(connectedEvent: ClientConnectionEstablishedEvent) {
        twitchSettings.initialChannels.map { asChannelName(it) }.forEach {
            connectedEvent.client.addChannel(it)
        }
    }
```

CAP, PING, NICK etc. messages will get sent, but the JOIN messages, or other non-immediate messages, never.

I doubt this is limited to just twitch, but the built-in twitch support requires an already created client. If instead I apply the support manually, with the only difference being providing the queue supplier in the builder, the messages get sent.

I tried debugging the issue, but it seemed like the sending queue (where the joins were) was properly processing the items yet for some reason they never ended up being written to the channel. I might be reading it wrong though.

**Comment by mbax** [comment]
Bug confirmed, working on finding it too. What a weird mess. :(

**Comment by mbax** [comment]
I think I found it. If I'm correct in what I'm interpreting: 

Even if the client hasn't started sending messages yet, replacing the sender will result in starting sending immediately, *before the connection even exists*, resulting in the messages never going over the non-established connection.

**Comment by mbax** [comment]
Potential fix here - https://github.com/KittehOrg/KittehIRCClientLib/commit/6af4c44ba55ceadeac1da5f6a5a4856751904305

I don't have the time to test it out tonight. If you get the opportunity, that commit is the only thing on 5.0.1-SNAPSHOT after 5.0.0 release.

**Comment by mbax** [comment]
Resolved in 5.0.1 which is released. I've learned a valuable lesson about testing before *and after* merging changes.
